dashboard/app/models/queued_account_purge.rb

```diff
@@ -43,4 +43,10 @@ def resolve!
     AccountPurger.new(bypass_safety_constraints: true).purge_data_for_account User.with_deleted.find(user_id)
     destroy!
   end
+
+  # It's possible to have a QueuedAccountPurge still around, pointing at an account that
+  # has already been purged.  This method finds and removes those records.
+  def self.clean_up_resolved_records!
+    all.select {|qap| User.with_deleted.find(qap.user_id).purged_at.present?}.each(&:destroy!)
```

Comment:
Not sure how this got lost the first time!

This is going to issue a select query for each QAP record, right? Is there a way to rewrite this to be more efficient, or do we not care so much about efficiency because we don't expect this to run very often?

Reply:
I think you're suggesting https://github.com/code-dot-org/code-dot-org/pull/31079/commits/979dc95825f1ec27589f02a02b59eff9c11d4464, and I _like_ it!   TIL you can remove the paranoia scope from an association.

Note that we're down to one SELECT query to find the records to delete, but still performing a DELETE query per record.  Turns out [`delete_all`](https://apidock.com/rails/v5.0.0.1/ActiveRecord/Relation/delete_all) sometimes doesn't play nice with `joins` and I'm not sure it's worth digging deep into that issue right now.

I'm not so worried about efficiency because

1. It runs once every night, close to nadir traffic.
2. The `all` set should always be small.  As of now, with two weeks of backlog to work through, it's below 150 records.

Reply:
That's exactly what I was thinking! I like it, too.

Good catch with the deletes; I hadn't even noticed that! I agree that it's probably find to not optimize further, but could you add a comment calling out why?